core/shared/src/main/scala/zio/prelude/ForEach.scala

```diff
@@ -301,6 +301,35 @@ trait ForEach[F[+_]] extends Covariant[F] { self =>
   def toList[A](fa: F[A]): List[A] =
     foldLeft(fa)(List.empty[A])((as, a) => a :: as).reverse
 
+  /**
+   * Zips the left collection and right collection together, using `None` to
+   * handle the case where one collection is larger than the other.
+   */
+  def zipAll[A, B, C](fa: F[A], fb: F[B])(implicit
+    both: IdentityBoth[F],
+    either: IdentityEither[F]
+  ): F[(Option[A], Option[B])] =
```

Comment:
Why not just `F[These[A, B]]` (instead of `F[(Option[A], Option[B])]`)?

Reply:
Good idea. I originally did it this way to mirror the signature of `zipAll` on `Chunk` but you're right that is a more precise signature.